src/bucket/LiveBucketIndex.cpp

```diff
@@ -95,6 +81,44 @@ LiveBucketIndex::LiveBucketIndex(BucketManager const& bm, Archive& ar,
     releaseAssertOrThrow(pageSize != 0);
 }
 
+void
+LiveBucketIndex::maybeInitializeCache(size_t bucketListTotalAccounts,
+                                      size_t maxBucketListAccountsToCache) const
+{
+    // Everything is already in memory, no need for a redundant cache, or we've
+    // already initialized this cache
+    if (mInMemoryIndex || mCache)
+    {
+        return;
+    }
+
+    releaseAssert(mDiskIndex);
+
+    auto accountsInThisBucket =
+        mDiskIndex->getBucketEntryCounters().entryTypeCounts.at(
+            LedgerEntryTypeAndDurability::ACCOUNT);
+
+    // Nothing to cache
+    if (accountsInThisBucket == 0)
+    {
+        return;
+    }
+
+    if (bucketListTotalAccounts < maxBucketListAccountsToCache)
+    {
+        // We can cache the entire bucket
+        mCache = std::make_unique<CacheT>(bucketListTotalAccounts);
+    }
+    else
+    {
+        double percentAccountsInBucket =
+            static_cast<double>(accountsInThisBucket) / bucketListTotalAccounts;
+        auto cacheSize = static_cast<size_t>(bucketListTotalAccounts *
```

Comment:
Currently `cacheSize == accountsInThisBucket`. You probably meant to use `maxBucketListAccountsToCache`.
Maybe it's worth adding a meter for this and writing a simple smoke test that ensures that the cache size is as expected given the BL size and cache size setting.

Reply:
Good catch, I've added robust unit tests now.